### PR TITLE
Provide an option to specify projection rebuild command timeout for Marten.CommandLine

### PR DESCRIPTION
**Pull request by mysticmind**
Fixes GH-2360

### CODE REVIEW

File: src/Marten.CommandLine/Commands/Projection/IProjectionHost.cs [hunk 1]
```diff
@@ -14,7 +16,7 @@ public interface IProjectionHost
 {
     IReadOnlyList<IProjectionStore> AllStores();
     void ListenForUserTriggeredExit();
-    Task<RebuildStatus> TryRebuildShards(IProjectionDatabase database, IReadOnlyList<AsyncProjectionShard> asyncProjectionShards);
+    Task<RebuildStatus> TryRebuildShards(IProjectionDatabase database, IReadOnlyList<AsyncProjectionShard> asyncProjectionShards, TimeSpan? shardTimeout=null);
```
Comment (jeremydmiller) [inline src/Marten.CommandLine/Commands/Projection/IProjectionHost.cs:19]: Ugh, I'm sorry, but we've gotta do that as an overload, not as an optional argument. That's actually a breaking change that would require a code compilation, even though the code itself wouldn't have to change

**Review by jeremydmiller — APPROVED** [review]
Works for me, but we better put a sample of the CLI usage in the docs later

**Review by oskardudycz — APPROVED** [review]

**Comment by NikiforovAll** [comment]
I would like to suggest reopening this story. Because the timespan "1h 30 mins" is not applied even when specified explicitly

@mysticmind, sorry if the screen doesn't tell all the required info, but I can assure you that it fails in 5 minutes or so.

image

cc: @jeremydmiller

**Comment by mysticmind** [comment]
@NikiforovAll thanks for the details, I will check on this and keep you posted.

**Pull request merged**